unittests/test_pipeline.py

```diff
@@ -525,6 +526,42 @@ def set_resources(self):
     assert expected_job_options == set(test.job.options)
 
 
+def test_pre_init_hook(local_exec_ctx):
+    with pytest.raises(ValueError):
+        @fixtures.custom_prefix('unittests/resources/checks')
+        class MyTest(rfm.RunOnlyRegressionTest):
+            @rfm.run_before('init')
+            def prepare(self):
+                self.x = 1
+
+
+def test_post_init_hook(local_exec_ctx):
+    class _T0(rfm.RunOnlyRegressionTest):
+        x = variable(int, value=0)
+        y = variable(int, value=1)
+
+        def __init__(self):
+            self.x = 1
+
+        @rfm.run_after('init')
+        def prepare(self):
+            self.y += 1
+
+    class _T1(_T0):
+        def __init__(self):
+            super().__init__()
+            self.z = 3
+
+    t0 = _T0()
+    assert t0.x == 1
+    assert t0.y == 2
+
+    t1 = _T1()
+    assert t1.x == 1
+    assert t1.y == 2
+    assert t1.z == 3
```

Comment:
What about concatenating strings here and then asserting for `t1.x == 'abc'`? That would test the order of execution too.

Reply:
Correct, I'll do that.